src/extractor/intersection/intersection_analysis.cpp

```diff
@@ -678,6 +679,7 @@ IntersectionView convertToIntersectionView(const util::NodeBasedDynamicGraph &gr
     BOOST_ASSERT(uturn.eid != SPECIAL_EDGEID);
     if (uturn.entry_allowed || allowed_uturns_number == 0)
     { // Add the true U-turn if it is allowed or no other U-turns found
+        BOOST_ASSERT(uturn.angle == 0.);
```

Comment:
`angle` is the perceived angle between the incoming and outgoing edges at the intersection. It can differ from the real angle because it does smart stuff like look further ahead down the road to understand where we're actually heading.

For the u-turn incoming/outgoing pair, any changes in angle will always match, so the difference remains `0.`. This is the important assumption that by adding it here guarantees we always have a 0° turn in the intersection view.